simulation/Scintillator_Energy/example/energy_simulation_example1.py

```diff
@@ -0,0 +1,223 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+#Define your parameters:
+
+#Length of orthogonal scintillators
+L=0.3
+
+#This is the maximum detectable angle in degrees,leads to the height between scint. layers by h=L/tan(θ_max)
+theta_max=60 
+
+#Minimum and maximum detectable muons energy(in GeV):
+E_min=1
+E_max=50
+
+#Number of possible energy values(this depends on SiPM's accuracy):
+N_E=6
+
+#For a random sample define 1,for a perfect CM distribution 0:
+random=1
+
+def MetropolisHastings(pdf, N, x_init, sigma):
+    chain_points=np.zeros((N,x_init.shape[0]))
+    stepsize=sigma
+    x_old=x_init
+    for i in range(N):
+      x_new =np.random.uniform(low=x_old-(1/2)*stepsize,high=x_old+(1/2)*stepsize,size=x_old.shape)
+      p_accept=min(1, pdf(x_new)/pdf(x_old))
+      accept = (np.random.random()<p_accept)
+      if accept:
+        x_old=x_new
+      chain_points[i,:]=x_old
+
+    return chain_points
+    
+def pdf_theta(x,theta_max=theta_max*np.pi/180):
+    if (x>theta_max) or (x<0):
+        return 0
+    return np.cos(x)**2
+
+def norm(theta,E_min=E_min,E_max=E_max,N_E=N_E):
+    dE=(E_max-E_min)/(N_E-1)
+    e_k=115
+    e_p=850
+    E=E_min
+    sum=0
+    for i in range (N_E):
+        sum+=0.14*dE*(E**(-2.7))*(((1/(1+1.1*E*np.cos(theta)/e_p))+(0.054/(1+1.1*E*np.cos(theta)/e_k))))
+        E+=dE
+    return sum
+
+
+def pdf_e(E,theta=0,E_min=E_min,E_max=E_max,n=N_E):
+    E_norm=norm(theta=theta)
+    if (E<E_min) or (E>E_max):
+        return 0
+    dE=(E_max-E_min)/(n-1)
+    e_k=115
+    e_p=850
+    p=0.14*dE*(E**(-2.7))*(((1/(1+1.1*E*np.cos(theta)/e_p))+(0.054/(1+1.1*E*np.cos(theta)/e_k))))
+    return p/E_norm
+    
+    
+def source(n_theta,n_phi,L,theta_max=60*np.pi/180,s_h=100):    
+   h=L/np.tan(theta_max)
+   z=np.ones((n_theta,n_phi))*(h/2+s_h)
+   x=np.zeros((n_theta,n_phi))
+   y=np.zeros((n_theta,n_phi))
+   directionx=np.zeros((n_theta,n_phi))
+   directiony=np.zeros((n_theta,n_phi))
+   directionz=np.zeros((n_theta,n_phi))
+   theta=np.linspace(0,theta_max,n_theta)
+   phi=np.linspace(0,2*np.pi-2*np.pi/n_phi,n_phi)
+
+   for i in range (n_theta):
+       R=z[0,0]*np.tan(theta[i])
+       for j in range (n_phi):
+           x[i,j]=R*np.cos(phi[j])
+           y[i,j]=R*np.sin(phi[j])
+           directionx[i,j]=-np.sin(theta[i])*np.cos(phi[j])
```

Comment:
can u add some comments? why =- and not -=?

Reply:
Yeah,I am going to send those to the others to check if they understand the whole process and if we need to add more details.The - in the direction comes from the fact that the source point position is always above the layers and target the (0,0,h/2).I will add a notebook that visualise the first functions,I think that's the easiest way to understand everything